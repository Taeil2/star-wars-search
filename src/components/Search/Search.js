import React from 'react';
import Context from './../Context/Context';

export default class Search extends React.Component {
  state = {
    character: '',
    error: ''
  }

  static contextType = Context;

  handleSubmit = (e) => {
    e.preventDefault();
    if (this.state.character === '') {
      this.setState({error: 'Enter a character name'})
    } else {
      this.setState({error: ''})
      this.context.runSearch(this.state.character);
    }
  };

  handleChange = (e) => {
    this.setState({character: e.target.value});
  }

  render() {
    return (
      <form id="searchForm" onSubmit={(e) => this.handleSubmit(e)}>
        <label htmlFor="character">Search for a character</label>
        <input id="character" type="text" placeholder="skywalker" onChange={this.handleChange}value={this.state.character}></input>
        <div className="error">{this.state.error}</div>
      </form>
    );
  }
}
