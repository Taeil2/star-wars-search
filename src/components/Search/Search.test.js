import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Search from './Search';
import './Search.scss';

// smoke test
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Search />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// snapshot test
it('renders the UI as expected', () => {
  const tree = renderer
    .create(<Search />)
    .toJSON();
  expect(tree).toMatchSnapshot();
  });
