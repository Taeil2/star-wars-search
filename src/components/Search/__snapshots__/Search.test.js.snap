// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form
  id="searchForm"
  onSubmit={[Function]}
>
  <label
    htmlFor="character"
  >
    Search for a character
  </label>
  <input
    id="character"
    onChange={[Function]}
    placeholder="skywalker"
    type="text"
    value=""
  />
  <div
    id="error"
  >
    
  </div>
</form>
`;
